rets/http/parsers/http.py

```diff
@@ -25,14 +25,29 @@ def parse_response(response: Response_) -> Any:
 def parse_xml(response: Response_) -> Element:
     root = XML(response.content)
 
-    reply_code = int(root.get('ReplyCode'))
-    if reply_code:
+    # if RETS-STATUS exists, the client must use this instead
+    # of the status from the body-start-line
+    rets_status = parse_rets_status(root)
+
+    if rets_status:
+        reply_code, reply_text = rets_status
+    else:
+        reply_code = int(root.get('ReplyCode'))
         reply_text = root.get('ReplyText')
+
+    if reply_code:
         raise RetsApiError(reply_code, reply_text, response.content)
 
     return root
 
 
+def parse_rets_status(elem: Element) -> Optional[Tuple[int, str]]:
+    rets_status = elem.find('RETS-STATUS')
+    if rets_status is None:
+        return None
```

Comment:
Maybe we can have `rets_status = elem` here so we can avoid the duplication with the root.get('ReplyCode'). We should probably rename the `elem -> root` as well.